superset/sql_lab.py

```diff
@@ -231,11 +231,12 @@ def handle_error(msg):
 
     # check whether the result set is comprised of lists or dict
     if data and len(data) > 0:
-        first_row_type = type(data[0])
-        if first_row_type is list or first_row_type is tuple:
-            df_data = np.array(data)
-        else:
+        first_row = data[0]
+        first_row_types = set([type(c) for c in first_row])
```

Comment:
Is this more readable?
```
any([isinstance(c, dict) for c in first_row])
```

Reply:
Yeah I just took a stab at refactoring for clarity